packages/material-ui-lab/src/internal/pickers/PickersToolbarText.tsx

```diff
@@ -1,46 +1,73 @@
-import { MuiStyles, StyleRules, WithStyles, withStyles } from '@material-ui/core/styles';
+import {
+  experimentalStyled,
+  unstable_useThemeProps as useThemeProps,
+} from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
 import { ExtendMui } from './typings/helpers';
+import pickersToolbarTextClasses, {
+  getPickersToolbarTextUtilityClass,
+} from './pickersToolbarTextClasses';
 
 export interface PickersToolbarTextProps extends ExtendMui<TypographyProps> {
+  component?: React.ElementType;
+  classes?: Partial<typeof pickersToolbarTextClasses>;
   selected?: boolean;
   value: React.ReactNode;
 }
 
-export type PickersToolbarTextClassKey = 'root' | 'selected';
-export const styles: MuiStyles<PickersToolbarTextClassKey> = (
-  theme,
-): StyleRules<PickersToolbarTextClassKey> => {
-  return {
-    root: {
-      transition: theme.transitions.create('color'),
-      color: theme.palette.text.secondary,
-      '&$selected': {
-        color: theme.palette.text.primary,
-      },
-    },
-    selected: {},
+export type PickersToolbarTextClassKey = keyof typeof pickersToolbarTextClasses;
+
+const useUtilityClasses = (styleProps: PickersToolbarTextProps) => {
+  const { selected, classes } = styleProps;
+
+  const slots = {
+    root: ['root', selected && 'selected'],
   };
-};
 
-const PickersToolbarText: React.FC<PickersToolbarTextProps & WithStyles<typeof styles>> = (
-  props,
-) => {
-  const { className, classes, selected, value, ...other } = props;
-
-  return (
-    <Typography
-      className={clsx(classes.root, className, {
-        [classes.selected]: selected,
-      })}
-      component="span"
-      {...other}
-    >
-      {value}
-    </Typography>
-  );
+  return composeClasses(slots, getPickersToolbarTextUtilityClass, classes);
 };
 
-export default withStyles(styles, { name: 'MuiPickersToolbarText' })(PickersToolbarText);
+const PickersToolbarTextRoot = experimentalStyled(
+  Typography,
+  {},
+  {
+    name: 'MuiPickersToolbarText',
+    slot: 'Root',
+    overridesResolver: (props, styles) => styles.root,
+  },
+)(({ theme }) => ({
+  transition: theme.transitions.create('color'),
+  color: theme.palette.text.secondary,
+  [`&.${pickersToolbarTextClasses.selected}`]: {
+    color: theme.palette.text.primary,
+  },
+}));
+
+const PickersToolbarText = React.forwardRef<HTMLSpanElement, PickersToolbarTextProps>(
+  function PickersToolbarText(inProps, ref) {
+    const props = useThemeProps({ props: inProps, name: 'MuiPickersToolbarText' });
+    const { component = 'span', className, selected, value, ...other } = props;
+    const styleProps = { ...props };
+
+    const classes = useUtilityClasses(styleProps);
+
+    return (
+      <PickersToolbarTextRoot
+        ref={ref}
+        className={clsx(classes.root, className)}
+        // TODO experimentalStyled should know component prop from Typography
```

Comment:
`experimentalStyled` will never be able to. We can't preserve generics in higher-order functions. Instead, the type of  `PickersToolbarTextRoot` should be cast accordingly.

Reply:
updated, not sure if this is okay or not.
```js
const PickersToolbarTextRoot = experimentalStyled<
  OverridableComponent<TypographyTypeMap<{ component?: React.ElementType }>>
>
```